### ISSUE TITLE
ZT tries to use internal address for P2P connection to remote node

### ISSUE THREAD

**Issue by StrikerTwo** [zerotier/ZeroTierOne#759]
For the last couple of days I have had weird connection issues. The connection to my central ZT node (running on a OPNsense VM) would drop every 5-10 seconds, then reconnect after 5-10 seconds, repeated endlessly.

A `listpeers` on my computer showed that the central node's IP address toggled between its WAN address (78.47.x.y) and its LAN address (10.8.a.b). The LAN address of course is only reachable over the ZT tunnel, so this simply could not work.

I seem to have solved it with a blacklist directive for 10.0.0.0/8 in the OPNsense local.conf, but shouldn't ZT notice stuff like that on its own? Trying to connect to a peer by an address that is only reachable over a ZT route to that same peer seems pretty stupid to me ;-)

This happened both on 1.2.4 and 1.2.8.

**Comment by dragon2611** [comment]
I think it tries to connect to ALL ip's on a machine, I have an OPNsense install with multiple public IP's present on it and I've seen Zerotier connect on any of the ones where port 9993 isn't forwarded elsewhere by a NAT rule.

**Comment by StrikerTwo** [comment]
Yes. That was my original problem, because connections would constantly swap between the available IP addresses. I solved that problem by setting up a separate OPNsense VM with only one public IP address, only ending up with this issue here.

Would be nice to have an option to choose which interfaces ZT binds to. 
Or at least if it must bind to all interfaces, it should select ONE address (prioritizing public IPs over RFC1918 ones) to report to ZT central AND STICK TO THAT.

**Comment by laduke** [comment]
@StrikerTwo Your OPNsense router is a Controller, and it's a different computer that is flipping between (78.47.x.y) and (10.8.a.b)?   


Side note: 0n linux, an alternative to a VM would be using a Network Namespace. 
2: There's also `interfacePrefixBlacklist` in local.conf

**Comment by StrikerTwo** [comment]
@laduke No, it's just one computer.

`interfacePrefixBlacklist` is what solved my problem with ZT announcing RFC1918 addresses, yes.

**Comment by StrikerTwo** [comment]
This is still a problem, see #779

**Comment by Crest** [comment]
Would it help to add the platform specific prefix common to all ZeroTier interfaces to `interfacePrefixBlacklist` in `local.conf` until this is fixed e.g. `feth*` on macOS and `zt*` on FreeBSD (when is this matched the tap interfaces are created as tapNNNN and renamed after all)?
